server/sonar-server/src/main/java/org/sonar/server/favorite/FavoriteUpdater.java

```diff
@@ -21,60 +21,58 @@
 package org.sonar.server.favorite;
 
 import java.util.List;
+import javax.annotation.Nullable;
 import org.sonar.db.DbClient;
 import org.sonar.db.DbSession;
 import org.sonar.db.component.ComponentDto;
 import org.sonar.db.property.PropertyDto;
 import org.sonar.db.property.PropertyQuery;
 import org.sonar.server.exceptions.BadRequestException;
-import org.sonar.server.user.UserSession;
 
 import static org.sonar.server.ws.WsUtils.checkRequest;
 
 public class FavoriteUpdater {
   static final String PROP_FAVORITE_KEY = "favourite";
 
   private final DbClient dbClient;
-  private final UserSession userSession;
 
-  public FavoriteUpdater(DbClient dbClient, UserSession userSession) {
+  public FavoriteUpdater(DbClient dbClient) {
     this.dbClient = dbClient;
-    this.userSession = userSession;
   }
 
   /**
-   * Set favorite to the logged in user. If no user is logged, no action is done
+   * Set favorite to the logged in user. If no user, no action is done
    */
-  public void add(DbSession dbSession, ComponentDto componentDto) {
-    if (!userSession.isLoggedIn()) {
+  public void add(DbSession dbSession, ComponentDto componentDto, @Nullable Long userId) {
```

Comment:
It's written in the description of the commit : having a UserSession as dependency of this class is forcing every classes using it to also have a UserSession => As it's now used in ComponentUpdater, which should not have a UserSession, it's better to use userId as it reduce dependencies